### PR TITLE
Upgrade dependencies (Next 12.1 and more)

### PR DESCRIPTION
**Pull request by kachkaev**
Methodology:

```sh
npx npm-check-updates --upgrade
rm -rf yarn.lock
yarn install
yarn fix:yarn-deduplicate
```

\+ see PR comments for changes in `next.config.mjs`

### CODE REVIEW

File: next.config.mjs [hunk 1]
```diff
@@ -18,12 +18,13 @@ process.env.NEXT_PUBLIC_SENTRY_ENVIRONMENT = process.env.SENTRY_ENVIRONMENT;
...
  */
 const nextConfig = {
-  experimental: {
+  compiler: {
     styledComponents: true,
   },
 
   productionBrowserSourceMaps: true,
   reactStrictMode: true,
+  swcMinify: true,
```
Comment (kachkaev) [inline next.config.mjs:27]: https://nextjs.org/blog/next-12-1#improved-swc-support

**Before / after**

Screenshot 2022-02-19 at 13 08 47

File: next.config.mjs [hunk 2]
```diff
@@ -18,12 +18,13 @@ process.env.NEXT_PUBLIC_SENTRY_ENVIRONMENT = process.env.SENTRY_ENVIRONMENT;
  * @todo Remove Omit<> when mismatch between Next Config and Sentry config is resolved
  */
 const nextConfig = {
-  experimental: {
+  compiler: {
```
Comment (kachkaev) [inline next.config.mjs:21]: https://nextjs.org/blog/next-12-1#improved-swc-support
https://nextjs.org/docs/advanced-features/compiler#styled-components

File: package.json [hunk 3]
```diff
@@ -28,52 +28,52 @@
   },
   "dependencies": {
     "@auth0/nextjs-auth0": "^1.7.0",
-    "@fortawesome/fontawesome-svg-core": "^1.2.36",
-    "@fortawesome/free-brands-svg-icons": "^5.15.4",
-    "@fortawesome/free-solid-svg-icons": "^5.15.4",
+    "@fortawesome/fontawesome-svg-core": "^1.3.0",
+    "@fortawesome/free-brands-svg-icons": "^6.0.0",
+    "@fortawesome/free-solid-svg-icons": "^6.0.0",
```
Comment (kachkaev) [inline package.json:33]: Breaking changes: https://fontawesome.com/docs/changelog/
(we are not affected)

File: package.json [hunk 4]
```diff
@@ -28,52 +28,52 @@
...
   },
   "devDependencies": {
     "@kachkaev/eslint-config-react": "^0.4.5",
     "@kachkaev/markdownlint-config": "^0.3.0",
-    "@netlify/plugin-nextjs": "^4.2.4",
-    "@next/eslint-plugin-next": "^12.0.10",
-    "@types/node": "^16.11.22",
+    "@netlify/plugin-nextjs": "^4.2.7",
+    "@next/eslint-plugin-next": "^12.1.0",
+    "@types/node": "^17.0.18",
```
Comment (kachkaev) [inline package.json:60]: We don’t use node directly, so the difference between 16 and 17 should not change anything for us.

**Review by sergous — APPROVED** [review]

**Pull request merged**